krm-functions/lib/kubeobject/kubeobject.go

```diff
@@ -79,9 +79,11 @@ func (o *KubeObjectExt[T1]) SetStatus(newStatus interface{}) error {
 	return safeSetNestedFieldKeepFormatting(&o.KubeObject, newStatus, "status")
 }
 
-// NOTE: the following functions are considered as "methods" of KubeObject,
-// and thus nill checking of `obj` was omitted intentionally:
-// the caller is responsible for ensuring that `obj` is not nil`
+// SetFromTypedObject sets the value of `o` to `value`, while keeping most of the YAML formatting.
+// It can be seen as an in-place version of fn.NewFromTypedObject
+func (o *KubeObjectExt[T1]) SetFromTypedObject(value T1) error {
+	return safeSetNestedFieldKeepFormatting(&o.KubeObject, value)
```

Comment:
I am thinking instead of doing  the following logic in  the method `safeSetNestedFieldKeepFormatting`
```

if len(fields) == 0 {
		obj2, err := fn.NewFromTypedObject(value)
		if err != nil {
			return err
		}
		*obj = *obj
```
Can we create the new object(obj2) in this method only, convert it to value equivalent  and pass  it to  `safeSetNestedFieldKeepFormatting`. That will keep that method consistent without any changes needed.

Does it make sense?

Reply:
> pass [the new object]  to `safeSetNestedFieldKeepFormatting`

what would you put into the `fields` parameter, when you passing the new object? 
In its current form if you call `safeSetNestedFieldKeepFormatting` without any fields parameter it will do nothing (this behavior is inherited from `obj.SetNestedField` that it calls internally.

Reply:
Well in `SetFromTypedObject` we will create the ivalue for the passes in go struct and simply call 
safeSetNestedFieldKeepFormatting(obj, value)
NO fields  passed i n and SetNestedField  take care of it.

Reply:
Unfortunately calling SetNestedField() with no fields at all is  a noop: it doesn't return an error, but it doesn't do anything at all either.

Reply:
isnt that what we want in this case as we are not passing any fields from  SetFromTypedObject?